File: Cpp/DSA/BinarySearch.cpp
#include <iostream>
using namespace std;

int main() {
    int n, key;
    cout<<"Enter the size of an array:";
    cin >> n;

    int arr[n];
    cout<<"Enter the element to an array:";
    for(int i = 0; i < n; i++)
        cin >> arr[i];
    cout<<"Enter the key ellement:";

    cin >> key;

    int low = 0, high = n - 1;

    while(low <= high) {
        int mid = (low + high) / 2;

        if(arr[mid] == key) {
            cout << "Found at index " << mid;
            return 0;
        }
        else if(arr[mid] < key)
            low = mid + 1;
        else
            high = mid - 1;
    }

    cout << "Not Found";
}

File: Java/DSA/BinarySearch.java
import java.util.*;

public class BinarySearch {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of an Array:");

        int n = sc.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements :");
        for(int i = 0; i < n; i++)
            arr[i] = sc.nextInt();
        System.out.println("Enter the key element:");
        int key = sc.nextInt();

        int low = 0, high = n - 1;
        boolean found = false;

        while(low <= high) {
            int mid = (low + high) / 2;

            if(arr[mid] == key) {
                System.out.println("Found at index " + mid);
                found = true;
                break;
            }
            else if(arr[mid] < key)
                low = mid + 1;
            else
                high = mid - 1;
        }

        if(!found)
            System.out.println("Not Found");
    }
}